syftlib/src/test/java/org/openmined/syft/monitor/network/NetworkStatusRealDataSourceTest.kt

```diff
@@ -0,0 +1,85 @@
+package org.openmined.syft.monitor.network
+
+import android.accounts.NetworkErrorException
+import android.content.Context
+import android.net.NetworkCapabilities
+import android.net.NetworkRequest
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.given
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.spy
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import io.reactivex.Completable
+import io.reactivex.Scheduler
+import io.reactivex.Single
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.processors.PublishProcessor
+import io.reactivex.schedulers.Schedulers
+import org.apache.tools.ant.taskdefs.condition.Http
+import org.junit.Test
+import org.openmined.syft.Syft
+import org.openmined.syft.domain.SyftConfiguration
+import org.openmined.syft.execution.JobStatusSubscriber
+import org.openmined.syft.execution.Plan
+import org.openmined.syft.execution.SyftJob
+import org.openmined.syft.monitor.DeviceMonitor
+import org.openmined.syft.monitor.StateChangeMessage
+import org.openmined.syft.networking.clients.HttpClient
+import org.openmined.syft.networking.clients.SocketClient
+import org.openmined.syft.networking.datamodels.ClientConfig
+import org.openmined.syft.networking.datamodels.syft.AuthenticationRequest
+import org.openmined.syft.networking.datamodels.syft.AuthenticationResponse
+import org.openmined.syft.networking.datamodels.syft.SpeedCheckResponse
+import org.openmined.syft.networking.requests.HttpAPI
+import org.openmined.syft.proto.SyftModel
+import org.openmined.syft.threading.ProcessSchedulers
+import retrofit2.Response
+import java.util.concurrent.ConcurrentHashMap
+
+@ExperimentalUnsignedTypes
+class NetworkStatusRealDataSourceTest {
+
+    private val workerId = "test id"
+    private val networkStatusModel = NetworkStatusModel()
+    private val constraints = listOf(
+        NetworkCapabilities.NET_CAPABILITY_INTERNET,
+        NetworkCapabilities.NET_CAPABILITY_NOT_RESTRICTED,
+        NetworkCapabilities.NET_CAPABILITY_NOT_METERED
+    )
+
+    private val networkStatusRealTimeDataSource = mock<NetworkStatusRealTimeDataSource> {
+        on { updatePing(workerId, networkStatusModel) }.thenReturn(Completable.complete())
+        on { updateDownloadSpeed(workerId, networkStatusModel) }.thenReturn(Completable.complete())
+        on { updateUploadSpeed(workerId, networkStatusModel) }.thenReturn(Completable.complete())
+        on { getNetworkValidity(constraints) }.thenReturn(true)
+    }
+
+    @Test
+    fun `network validity should return true if network constraints satisfied`() {
+        assert(networkStatusRealTimeDataSource.getNetworkValidity(constraints))
```

Comment:
This test is bound to succeed always since in stubbing you have asked the method to always return true

Reply:
One way to test whether the networkValidity works properly is by setting up the relevant network as active on the context. Robolectric might come on handy over here
Check the file in integration/AbstractSyftWorkerTest